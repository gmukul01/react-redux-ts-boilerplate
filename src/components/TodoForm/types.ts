import { AddTodoAction } from '@store/todos';

export interface Props {
    addTodo: (value: string) => AddTodoAction;
}
