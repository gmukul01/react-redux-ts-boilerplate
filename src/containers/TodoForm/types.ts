import { AddTodoAction } from '@store/todos';

export interface DispatchProps {
    addTodo: (text: string) => AddTodoAction;
}

// We can merge all props here StateProps & DispatchProps & OwnProps and use this in the component
export type Props = DispatchProps;
