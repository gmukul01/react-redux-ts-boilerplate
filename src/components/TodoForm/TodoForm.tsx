import Button from '@components/Button';
import Input from '@components/Input';
import React, { FormEvent, useRef } from 'react';
import { Props } from './types';

const TodoForm: React.FunctionComponent<Props> = ({ addTodo }) => {
    const inputEl = useRef<HTMLInputElement>(null);

    const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
        e.preventDefault();

        const value = inputEl.current.value.trim();

        if (value) {
            addTodo(value);
            inputEl.current.value = '';
        }
    };

    return (
        <form onSubmit={handleSubmit}>
            <Input ref={inputEl} placeholder="Enter Todo" />
            <Button solid type="submit">
                Add Todo
            </Button>
        </form>
    );
};

export default TodoForm;
