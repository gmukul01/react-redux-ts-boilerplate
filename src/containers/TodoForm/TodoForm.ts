import { TodoForm } from '@components';
import { addTodo } from '@store/todos';
import { connect } from 'react-redux';
import { DispatchProps } from './types';

const mapDispatchToProps: DispatchProps = {
    addTodo
};

export default connect<{}, DispatchProps>(
    null,
    mapDispatchToProps
)(TodoForm);
