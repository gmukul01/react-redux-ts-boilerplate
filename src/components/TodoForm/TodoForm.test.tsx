import 'jest-styled-components';
import React from 'react';
import { fireEvent, render } from 'react-testing-library';

import TodoForm from './TodoForm';

describe('AddTodo component', () => {
    const mockAddTodo = jest.fn();

    it('should render', () => {
        const { container } = render(<TodoForm addTodo={mockAddTodo} />);
        expect(container).toMatchSnapshot();
    });

    it('should call addTodo prop on click of add todo button', () => {
        const { getByPlaceholderText, getByText } = render(<TodoForm addTodo={mockAddTodo} />),
            addTodoInput = getByPlaceholderText('Enter Todo'),
            submitButton = getByText('Add Todo');

        fireEvent.change(addTodoInput, { target: { value: 'Random' } });
        fireEvent.submit(submitButton);
        expect(mockAddTodo).toHaveBeenCalledWith('Random');
    });
});
