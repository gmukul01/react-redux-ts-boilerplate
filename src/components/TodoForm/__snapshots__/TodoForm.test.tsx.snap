// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoForm component should render 1`] = `
.c2 {
  font-size: 14px;
  font-weight: 700;
  color: inherit;
  text-transform: uppercase;
}

.c1 {
  background-color: transparent;
  border: none;
  padding: 10px;
  margin: 8px;
  border-radius: 3px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #5392f9;
  color: #ffffff;
}

.c1:hover {
  cursor: pointer;
}

.c1:focus {
  outline: none;
}

.c0 {
  padding: 10px;
  font-size: 14px;
  font-weight: 400;
  color: #2a2a2e;
  background-color: #ffffff;
  background-clip: padding-box;
  box-sizing: border-box;
  border: 1px solid #dddfe2;
  border-radius: 3px;
}

.c0:focus {
  border-color: #5392f9;
  outline: 0;
}

.c0::-webkit-input-placeholder {
  color: #d7d7db;
}

.c0::-moz-placeholder {
  color: #d7d7db;
}

.c0:-ms-input-placeholder {
  color: #d7d7db;
}

.c0::placeholder {
  color: #d7d7db;
}

<div>
  <form>
    <input
      class="c0"
      placeholder="Enter Todo"
    />
    <button
      class="c1"
      type="submit"
    >
      <strong
        class="c2"
      >
        Add Todo
      </strong>
    </button>
  </form>
</div>
`;
